config/log_config.go

```diff
@@ -0,0 +1,77 @@
+package config
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"go.viam.com/rdk/logging"
+)
+
+// LoggerPatternConfig is an instance of a level specification for a given logger.
+type LoggerPatternConfig struct {
+	Pattern string `json:"pattern"`
+	Level   string `json:"level"`
+}
+
+const (
+	validLoggerSectionName             = `[a-zA-Z0-9]+([_-]*[a-zA-Z0-9]+)*`
+	validLoggerSectionNameWithWildcard = `(` + validLoggerSectionName + `|\*)`
+	validLoggerSections                = validLoggerSectionNameWithWildcard + `(\.` + validLoggerSectionNameWithWildcard + `)*`
+	validLoggerName                    = `^` + validLoggerSections + `$`
+)
+
+var loggerPatternRegexp = regexp.MustCompile(validLoggerName)
+
+func validatePattern(pattern string) bool {
+	return loggerPatternRegexp.MatchString(pattern)
+}
+
+// UpdateLoggerRegistry updates the logger registry if necessary  with the specified logConfig.
+func UpdateLoggerRegistry(logConfig []LoggerPatternConfig, loggerRegistry map[string]logging.Logger) (map[string]logging.Logger, error) {
+	newLogRegistry := make(map[string]logging.Logger)
+
+	for _, lpc := range logConfig {
+		if !validatePattern(lpc.Pattern) {
+			return nil, errors.New("Failed to Validate a Pattern")
+		}
+
+		var matcher strings.Builder
+		for idx, ch := range lpc.Pattern {
+			switch ch {
+			case '*':
+				if idx == len(lpc.Pattern)-1 {
+					matcher.WriteString(validLoggerSections)
+				} else {
+					matcher.WriteString(validLoggerSectionNameWithWildcard)
+				}
+			case '.':
+				matcher.WriteString(`\.`)
+			default:
+				matcher.WriteRune(ch)
+			}
+		}
+		r := regexp.MustCompile(matcher.String())
```

Comment:
This will panic if the regex does not compile. Panicking at compile time with the `MustCompile` you have at the top of the file is probably fine, but doing so at runtime would crash the RDK if a user used a bad regex. Can we use `Compile` instead and return any created error?